tools/system_libs.py

```diff
@@ -17,10 +17,20 @@
 import zipfile
 from glob import iglob
 
-from . import ports
 from . import shared
 from tools.shared import mangle_c_symbol_name, demangle_c_symbol_name
 
+try:
```

Comment:
Ooo.. thats a little grosser than I thought it would be... Seems like it would be nicer to be data driven.  

I'd like to take stab at doing this more cleanly.. and actually scanning for available ports, which it doesn't look like ti does currently.

If you want to land this in the mean time can you at add a `TODO(sbc):` to improve it?

Reply:
I don't know that it's super-urgent - can wait for you to take a look if you prefer a cleaner approach?

(Personally, this looks like a clean approach to me.. it shims the ports import. But I know you have stronger feelings on Python style and your opinions are probably more correct.)

Reply:
For one thing the commend you added is not true is it?  Adding an removing ports is not as simple as just adding files, you need to modify the __init__.py for each port you add?

Reply:
Oh, yes, we should make `__init__` scan for files I guess, good point.